externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/ui/EngineTab.scala

```diff
@@ -107,4 +122,10 @@ case class EngineTab(
       engine.get.stop()
     }
   }
+
+  def handleGracefulKillRequest(request: HttpServletRequest): Unit = {
+    if (killEnabled && engine.isDefined && engine.get.getServiceState != ServiceState.STOPPED) {
+      engine.get.gracefulStop()
```

Comment:
what happens if user multiple clicks `gracefulStop` ?

Reply:
Because the service can only be stopped once, when the service is in the stop state, multiple clicks will not affect.

## Engine log
### First click
```
INFO EngineServiceDiscovery: Service[EngineServiceDiscovery] is stopped.
```
## Second click
```
WARN EngineServiceDiscovery: Service[EngineServiceDiscovery] is not started(STOPPED) yet.
```